src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlImplTest.java

```diff
@@ -64,48 +77,98 @@
     @Mock
     private MultiPathDecorator tailDecorator;
 
+    @Mock
+    private WiresMagnet headMagnet;
+
+    @Mock
+    private WiresMagnet tailMagnet;
+
     private WiresManager wiresManager;
     private Layer layer;
     private PolyLine line;
     private WiresConnector connector;
-    private Point2D CP0 = new Point2D(0, 50);
-    private Point2D CP1 = new Point2D(50, 50);
-    private Point2D CP2 = new Point2D(100, 50);
+    private Point2D CP0;
+    private Point2D CP1;
+    private Point2D CP2;
+    private Point2D CP3;
+    private Point2D CP4;
 
     private WiresConnectorControlImpl tested;
 
+    @Mock
+    private IPrimitive headPrimitive;
+
+    @Mock
+    private IPrimitive tailPrimitive;
+
+    @Mock
+    private MagnetManager.Magnets magnets;
+
+    @Mock
+    private WiresShape shape;
+
     @Before
     public void setup() {
+        CP0 = new Point2D(CP0_INIT);
+        CP1 = new Point2D(CP1_INIT);
+        CP2 = new Point2D(CP2_INIT);
+        CP3 = new Point2D(CP3_INIT);
+        CP4 = new Point2D(CP4_INIT);
         layer = new Layer();
         wiresManager = WiresManager.get(layer);
         wiresManager.setControlPointsAcceptor(IControlPointsAcceptor.ALL);
         when(wiresLayer.getLayer()).thenReturn(layer);
         when(headDecorator.getPath()).thenReturn(headPath);
         when(tailDecorator.getPath()).thenReturn(tailPath);
-        CP0 = new Point2D(0, 50);
-        CP1 = new Point2D(50, 50);
-        CP2 = new Point2D(100, 50);
-        line = new PolyLine(CP0, CP1, CP2);
-        connector = spy(new WiresConnector(line,
-                                           headDecorator,
-                                           tailDecorator));
+        when(headMagnet.getControl()).thenReturn(headPrimitive);
+        when(tailMagnet.getControl()).thenReturn(tailPrimitive);
+        when(headPrimitive.getComputedLocation()).thenReturn(CP0);
+        when(tailPrimitive.getComputedLocation()).thenReturn(CP4);
+        when(headMagnet.getMagnets()).thenReturn(magnets);
+        when(tailMagnet.getMagnets()).thenReturn(magnets);
+        when(headMagnet.getIndex()).thenReturn(1);
+        when(tailMagnet.getIndex()).thenReturn(1);
+        when(magnets.getWiresShape()).thenReturn(shape);
+        when(shape.getPath()).thenReturn(shapePath);
+
+        line = spy(new PolyLine(CP0, CP1, CP2, CP3, CP4));
+
+//        connector = spy(new WiresConnector(line,
```

Comment:
I would remove code, which is commented out or add comment, why it is commented out.